src/components/List.js

```diff
@@ -19,6 +19,78 @@ You cannot undo this action, and this item's purchase history will be lost.`,
     return response ? deleteItem(token, result.id) : null;
   }
 
+  function predictedNextPurchase(lastPurchase, frequency) {
+    //takes the last purchase date, frequency, and current date and calculates how many days are there until anticipated purchase date
+    return Math.round((lastPurchase + frequency - Date.now()) / 86400000);
+  }
+
+  function lastPurchaseDate(result) {
+    // needs to be connected to itemAdded timestamp from other branch
+    //placeholder for addedItem date
+    const placeholderAddedDate = new Date('September 3, 20 00:20:18');
+    //if there is a purchase date, it's returning the recent purchase date. Otherwise, returning addedDate
+    return result.purchaseDates.length > 0
+      ? new Date(Math.max(...result.purchaseDates)).getTime()
+      : placeholderAddedDate.getTime();
+  }
+
+  function sortedResults() {
+    let active = [];
+    let inactive = [];
+    results.forEach(result => {
+      //pulls out inactive items - identified by having a negative number less than the negative frequency, or has a frequency greater than 30.
+      //for example, if an item has a frequency of 7 days, and the anticpated days left is -8, that means it's been 15 days since the last purchase date, and is inactive
+      if (
+        predictedNextPurchase(lastPurchaseDate(result), result.frequency) <
+          -1 * Math.round(result.frequency / 86400000) ||
+        Math.round(result.frequency / 86400000) > 30
+      ) {
+        result.timeClass = 'inactive';
+        inactive.push(result);
```

Comment:
The inactive AC wasn't part of the issue when I participated in the program, so really cool to see your approach to tackling this one. Great job!